yinsolidated/plugin/plugin.py

```diff
@@ -288,7 +285,45 @@ def _make_extension_element(statement, parent_elem):
     nsmap = {prefix: namespace}
 
     extension_element = etree.SubElement(parent_elem, tag, nsmap=nsmap)
-    extension_element.text = statement.arg
+
+    if _is_complex_extension(statement.i_extension):
+        argument_name, is_arg_yin_element = _get_exension_argument_details(
+            statement.i_extension
+        )
+        _add_statement_argument(
+            argument_name,
+            statement.arg,
+            namespace,
+            is_arg_yin_element,
+            extension_element,
+        )
+        _append_children(statement, extension_element)
```

Comment:
Will this recursively add other children (aka can we do deeply nested extensions with this change)? I don't think we have a need at the moment but we may want it in the future.

Reply:
Yup, this is recursive.